src/tests/multihost/alltests/test_ldap_extra_attrs.py

```diff
@@ -193,3 +195,53 @@ def test_0006_bz1667252(self, multihost):
         cmd = multihost.client[0].run_command(sssctl_cmd)
         ret = multihost.client[0].service_sssd('status')
         assert cmd.returncode == 0 and ret == 0
+
+    @staticmethod
+    @pytest.mark.tier1
+    def test_bz847043(multihost, backupsssdconf):
+        """
+        :title: Thread issue can cause the application to not get
+            any identity information bz847043
+        :id: a3f5b5ea-9cc6-11ed-98f2-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=847043
+        :steps:
+            1. rm -f /var/lib/sss/mc/*
+            2. log in via the SSSD
+            3. compile the provided C test program and run it
+            4. with the fixed version, the program should run to completion:
+                ./client-hang
+                Cancelling thread
+                Joining...
+                Joined, trying getpwuid_r call
+                Never get here
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+            4. Should succeed
+        """
+        tools = sssdTools(multihost.client[0])
+        client = multihost.client[0]
+        tools.sssd_conf("nss", {'filter_groups': 'root',
+                                'filter_users': 'root'}, action='update')
+        tools.sssd_conf("domain/example1", {'use_fully_qualified_names': False}, action='update')
+        tools.clear_sssd_cache()
+        file_location_c = '/script/sssd_client_hang.c'
+        client.transport.put_file(os.path.dirname(os.path.abspath(__file__))
+                                  + file_location_c, '/tmp/sssd_client_hang.c')
+        client.run_command("touch /tmp/output")
+        client.run_command("chown foo1 /tmp/output")
+        ssh = pxssh.pxssh(options={"StrictHostKeyChecking": "no", "UserKnownHostsFile": "/dev/null"})
+        ssh.force_password = True
+        try:
+            ssh.login(multihost.client[0].sys_hostname, 'foo1', 'Secret123')
+            ssh.sendline("cd /tmp")
+            ssh.prompt(timeout=5)
+            ssh.sendline('gcc -lpthread sssd_client_hang.c -o client-hang')
```

Comment:
Please run the compilation outside of expect and change ownership of the compiled binary.
Debugging issues in expect is painful.